### PR TITLE
Compressor redux

### PR DESCRIPTION
**Pull request by cantsin**
[Link to ZenHub issue.](https://app.zenhub.com/workspaces/doj-crt-intake-scrum-board-5d03bf56c1c8a35d482eca0f/issues/18f/crt-portal/792)

## What does this change?

Second try at implementing django-compressor functionality. Believe it or not, the `hero.png` image (which we never use) is indeed needed and breaks [manifest storage](https://docs.djangoproject.com/en/2.2/ref/contrib/staticfiles/#manifeststaticfilesstorage) outright if we omit this image. Why? Well, the USWDS CSS specifies `background-image:url("../img/hero.png");` which the Django manifest storage backend picks up and then subsequently crashes during `collectstatic` when this file is not found. tl;dr: I believe this missing, unused file is what broke deployment last time (really). 

The rest is identical to before, except for a couple of small things. We can use the built in `S3ManifestStaticStorage` instead of monkey patching classes. Two CSS `url` references have been tweaked so that they are properly fetched from S3. 

But most exciting is the `localstack` documentation I've thrown in, which explains how to set up an identical-ish production environment locally.

### Author

+ [x] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [x] Check for, document, and establish a testing plan for any behavior that may vary across environments or is otherwise difficult to test.
+ [x] Check for [accessibility](/docs/a11y_plan.md).
+ [x] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

### Reviewer

+ [ ] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [ ] Check for any behavior that may vary across environments or is difficult to test, and ensure that it is well-understood, documented, and that there is a testing plan in place.
+ [ ] Re-check for [accessibility](/docs/a11y_plan.md).
+ [ ] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

## Notes for reviewer:

See [PR instructions doc](https://github.com/usdoj/crt-portal/blob/master/docs/pull_requests.md) for full pull request review instructions.

### CODE REVIEW

**Comment by cantsin** [comment]
@Jkrzy flagging you as a reviewer too -- oddly enough Github is not populating your username in the reviewers field for whatever reason.

**Comment by cantsin** [comment]
Anyway, I'd like one of y'all to go through the documentation and let me know what worked, what was unclear, what could be improved. Thanks!

**Comment by Jkrzy** [comment]
This is :amaze:ing @cantsin!!! 

Running through the docs I get tripped up on `collectstatic`  with:

`s3_1   | 2021-01-07T15:23:09:INFO:localstack.services.edge: Unable to find forwarding rule for host "s3:4566"`

and this exception from boto: `botocore.exceptions.ClientError: An error occurred (404) when calling the PutObject operation: Not Found`

Did you hit this in local testing?

**Comment by cantsin** [comment]
@Jkrzy I have not gotten that error! To double check the obvious, did you rebuild the docker-compose image? I just ran through the steps again myself and so far I don't see that error. 

- Does `http://localhost:4566/health?reload` show the s3 service running? 
- If you do `docker-compose run web curl http://s3:4566/health` what shows up?

These two test external and internal access to the localstack edge port, respectively.

**Comment by cantsin** [comment]
@Jkrzy updated to address the issue we found last Friday. Please note that you will need to remove your local S3 bucket and re-create it (since `collectstatic` will skip if the file is already on S3 and therefore not save to `staticfiles` in the container as per our new storage class). Alternatively, I think removing the files: `aws s3 rm --recursive s3://crt-portal/ --endpoint-url=http://localhost:4566` will also work.

Tested this time by doing `git clean -fxd` beforehand :grinning:

File: crt_portal/crt_portal/storage.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from django.core.files.storage import get_storage_class
+from storages.backends.s3boto3 import S3Boto3Storage
+
+
+class CachedS3Boto3Storage(S3Boto3Storage):
```
Comment (Jkrzy) [inline crt_portal/crt_portal/storage.py:5]: I think we want `S3ManifestStaticStorage` to keep the file hashing/cache breaking.
```suggestion
from storages.backends.s3boto3 import S3ManifestStaticStorage


class CachedS3Boto3Storage(S3ManifestStaticStorage):
```

File: crt_portal/crt_portal/storage.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    """
+    Save compressed files to local storage before uploading to S3.
+    """
+    def __init__(self, *args, **kwargs):
+        super(CachedS3Boto3Storage, self).__init__(*args, **kwargs)
+        self.local_storage = get_storage_class("compressor.storage.CompressorFileStorage")()
+
+    def save(self, name, content):
+        self.local_storage._save(name, content)
+        super(CachedS3Boto3Storage, self).save(name, self.local_storage._open(name))
```
Comment (Jkrzy) [inline crt_portal/crt_portal/storage.py:15]: Quick fix to update this usage of super
```suggestion
        super().__init__(*args, **kwargs)
        self.local_storage = get_storage_class("compressor.storage.CompressorFileStorage")()

    def save(self, name, content):
        self.local_storage._save(name, content)
        super().save(name, self.local_storage._open(name))
```

**Review by Jkrzy — CHANGES_REQUESTED** [review]
Thanks @cantsin! We're super close but this latest change reverts to the non-manifesting storage implementation.

File: crt_portal/crt_portal/storage.py (hunk 2 above)
Comment (cantsin) [reply crt_portal/crt_portal/storage.py]: Thanks for both suggestions! I had `S3ManifestStaticStorage` locally but did not commit it :man_facepalming: Going through the paces once more to make sure everything is kosher.

**Review by Jkrzy — APPROVED** [review]
Looks good, thanks for sticking with this one @cantsin!

**Pull request merged**